streaming/src/main/scala/org/apache/spark/streaming/Checkpoint.scala

```diff
@@ -43,10 +43,15 @@ class Checkpoint(@transient ssc: StreamingContext, val checkpointTime: Time)
   val delaySeconds = MetadataCleaner.getDelaySeconds(ssc.conf)
   val sparkConfPairs = ssc.conf.getAll
 
-  def sparkConf = {
-    new SparkConf(false).setAll(sparkConfPairs)
+  def createSparkConf(): SparkConf = {
+    val newSparkConf = new SparkConf(loadDefaults = false).setAll(sparkConfPairs)
       .remove("spark.driver.host")
       .remove("spark.driver.port")
+    new SparkConf(loadDefaults = true).getOption("spark.master") match {
+      case Some(newMaster) => newSparkConf.setMaster(newMaster)
+      case _ => None
```

Comment:
`case _ => ` should be enough, no? Is there a need to return `None`? (Since `None` is an object, I am not sure what the real cost is in this case for returning something - so this can be ignored if the cost here is ~zero)